var/spack/repos/builtin/packages/bamtools/package.py

```diff
@@ -36,11 +37,14 @@ class Bamtools(Package):
     version('2.3.0', 'd327df4ba037d6eb8beef65d7da75ebc')
     version('2.2.3', '6eccd3e45e4ba12a68daa3298998e76d')
 
-    depends_on('cmake', type='build')
+    depends_on('zlib', type='link')
 
-    def install(self, spec, prefix):
-        with working_dir('spack-build', create=True):
-            cmake('..', *std_cmake_args)
+    def cmake_args(self):
+        args = []
 
-            make()
-            make('install')
+        rpath = self.rpath
+        rpath.append(os.path.join(self.prefix.lib, "bamtools"))
+        args.extend(std_cmake_args)
+        args.append("-DCMAKE_INSTALL_RPATH=%s" % ':'.join(rpath))
```

Comment:
That would be fine, but that appears to be a property the returns a static method in the CMakePackage class, which is why I went with cmake_args. Now that I know that the standard cmake arguments are included already, I can just add the one extra which will be appended to the end of the argument string. Since the **cmake_args** variable is guaranteed to be at the end of the string, there should be no issue overriding **-DCMAKE_INSTALL_PATH**.

This seems (to me) like the best compromise for now.

Ultimately, I think that the better "fix" would be to allowing the package (and package maintainer) to specify the expected library path for a package if it differs from the norm. This might save a lot of trouble down the road. A lot of software used in bioinformatics and computational biology isn't built with standardization in mind. 

We will need to make a similar change for any of the software that tries to link against the libbamtools library -- there are a fair amount that my group is working on for the near future.

I'll merge in these change shortly.